### PR TITLE
revert changes from merge of develop into sls_app.

### PR DESCRIPTION
**Pull request by raghits**
### Requirements

* Changes in service-onboarding-build-pack
* Changes in metrics API
* Changes in create-serverless-service (minor)

### Description of the Change

Backend Changes that were inadvertantly reverted during merge of develop into sls_app

### Benefits

Restore issues

### Possible Drawbacks

None

### Applicable Issues

None

### CODE REVIEW

File: builds/service-onboarding-build-pack/Jenkinsfile [hunk 1]
```diff
@@ -133,6 +135,11 @@ node  {
 				service_template = "${serviceType}-template-go"
 			}
 		}
+		} else {
```
Comment (devsatishm) [inline builds/service-onboarding-build-pack/Jenkinsfile]: Looks like a merge issue again - 
- Two events - one  with Success and another with Failure status in the code block.
- `else` block looks misplaced. Please check this again.

File: builds/service-onboarding-build-pack/Jenkinsfile [hunk 2]
```diff
@@ -160,7 +167,7 @@ node  {
                 def serviceMetadataJson = [:]
                 if(service_config['provider'] == 'aws'){
 					def roleARN
-                    if (service_config['domain'] == "jazz") {
+                    if (service_config['domain'] && service_config['domain'] == "jazz") {
```
Comment (devsatishm) [inline builds/service-onboarding-build-pack/Jenkinsfile]: service_config['domain'] is a required field now. Check not required.

File: core/jazz_create-serverless-service/index.js [hunk 3]
```diff
@@ -92,7 +92,7 @@ var handler = (event, context, cb) => {
     // Validate and set deployment accounts
     let primaryAccountCount = 0;
 
-    if (Array.isArray(service_creation_data.deployment_accounts) && service_creation_data.deployment_accounts) {
+    if (service_creation_data.deployment_accounts && Array.isArray(service_creation_data.deployment_accounts) && service_creation_data.deployment_accounts) {
```
Comment (devsatishm) [inline core/jazz_create-serverless-service/index.js]: Remove `service_creation_data.deployment_accounts` null check at the end.

File: core/jazz_metrics/components/utils.js [hunk 4]
```diff
@@ -200,7 +200,7 @@ function getAssetsObj(assetsArray, userStatistics) {
     } else {
       // type not found
       newAssetArr.push({
-        "message": `Asset type not found `,
+        "message": `Asset type not found: ${assetType}`,
```
Comment (devsatishm) [inline core/jazz_metrics/components/utils.js]: Isn't `assetType` null now? Why log it explicitly as null again?

File: builds/service-onboarding-build-pack/Jenkinsfile (hunk 1 above)
Comment (raghits) [reply builds/service-onboarding-build-pack/Jenkinsfile]: Addressed this.

File: builds/service-onboarding-build-pack/Jenkinsfile (hunk 2 above)
Comment (raghits) [reply builds/service-onboarding-build-pack/Jenkinsfile]: not needed. so removed it.

File: core/jazz_metrics/components/utils.js (hunk 4 above)
Comment (raghits) [reply core/jazz_metrics/components/utils.js]: No, `assetType` is required. We get metrics for all assets of a given assetType. Asset Name is not passed to the API.

File: core/jazz_create-serverless-service/index.js (hunk 3 above)
Comment (raghits) [reply core/jazz_create-serverless-service/index.js]: Array.isArray check is needed only if `service_creation_data.deployment_accounts` is not null - removing null check at the end.

File: core/jazz_metrics/components/utils.js (hunk 4 above)
Comment (devsatishm) [reply core/jazz_metrics/components/utils.js]: Hmm, do you need this `else` block  then?

File: core/jazz_services/components/crud/getList.js [hunk 5]
```diff
@@ -123,11 +123,11 @@ module.exports = (query, servicesList, onComplete) => {
               items_formatted = utils.filterUtil(items_formatted, query.filter);
             }
 
-            count = items_formatted.length;
             if (query.limit && query.offset) {
               items_formatted = utils.paginateUtil(items_formatted, parseInt(query.limit), parseInt(query.offset));
             }
 
+            count = items_formatted.length;
```
Comment (devsatishm) [inline core/jazz_services/components/crud/getList.js]: Revert this change.

File: core/jazz_metrics/components/utils.js (hunk 4 above)
Comment (raghits) [reply core/jazz_metrics/components/utils.js]: Removed else block and changed the last else if to else.

File: core/jazz_services/components/crud/getList.js (hunk 5 above)
Comment (raghits) [reply core/jazz_services/components/crud/getList.js]: Reverted.
